### ISSUE TITLE
Enable release notes automation

### ISSUE THREAD

**Issue by jsafrane** [kubernetes-csi/docs#142]
It's hard to keep track of changes between releases and compose correct CHANGELOG.

Proposal:
* Add PR template to all kubernetes-csi repos. Template of the template: https://github.com/kubernetes-csi/external-attacher/pull/139. It's slightly trimmed version of kubernetes/kubernetes template with removed links to Kubernetes specific docs.

* Enable release-notes bot in prow that will nag people to fill ` ```release-note` (same as in kubernetes/kubernetes).

* PR reviewers/approvers must ensure that ` ```release-note ` is filled in PRs where is makes sense.

* Use [release-notes tool](https://github.com/kubernetes/release/tree/master/cmd/release-notes) to generate CHANGELOG from ` ```release-note`, ` ```release-note Action required:` and `/kind {feature|bug}` in PRs. See [Kubernetes 1.11](https://gist.github.com/marpaia/acfdb889f362195bb683e9e09ce196bc) - we won't get SIG sections, but the rest should be similar.  Some manual editing is still required (deprecations?).

@kubernetes-csi/csi-misc

**Comment by saad-ali** [comment]
Seems like a great idea

**Comment by msau42** [comment]
Here's one sample invocation:
```
./release-notes --start-sha=0ed6978fd199e3ca10326b82b4b8b8e916211c9b --end-sha=3cb3d2f18ed8cb40371c6d8886edcabd1f27e7b9 --github-org=kubernetes-csi --github-repo=external-attacher -branch=master -output out.md
```

Would be nice to potentially add something like this into our release tools repo and have it pick up from a particular tag.

**Comment by jsafrane** [comment]
What's remaining to be done? cluster-driver-registrar is deprecated and the other repos have a PR template.

**Comment by msau42** [comment]
I think having a script in csi-release-tools to generate the params to the tool would be nice

**Comment by msau42** [comment]
Something doesn't seem right right the release tool generator. When I ran:
```
release-notes --start-sha=0bee749334d32e9e07cce2aaa90bbdba9a12d04a --end-sha=a1a8f48aa47c3eb63d108f4245b88d243601a90e --github-org=kubernetes-csi --github-repo=external-provisioner -branch=master -output out.md
```

it picked up a bunch of release notes from commits that we already released in 1.3.0

**Comment by davidz627** [comment]
This seems mostly resolved. I believe the remaining work is to make this more automated with a script.
